clients/go/pkg/vexilla/client.go

```diff
@@ -125,10 +126,11 @@ func (client *Client) GetFlags(groupNameOrId string, fetch func(url string) (*ht
 
 // Sets a fetched flag group within the Client instance.
 func (client *Client) SetFlags(group Group) error {
+
 	groupId := client.groupLookupTable[string(group.GroupId)]
 
 	if groupId == "" {
-		return fmt.Errorf("no group found for: %s", group.GroupId)
+		return NewGroupLookupError("")
```

Comment:
suggestion: Looks like this should probably be the group.GroupId value rather than an empty string

Reply:
Keen eye, added the group.GroupId here!